superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx

```diff
@@ -115,78 +121,90 @@ const CredentialsInfo = ({
             placeholder="Paste content of service credentials JSON file here"
           />
           <span className="label-paste">
-            {t('Copy and paste the entire service account .json file here')}
+            {t(
+              'Copy and paste the entire service account .json file here to enable connection',
```

Comment:
We should reuse the same component we have for uploading/pasting credentials in BQ.

Reply:
Also, I'm a bit worried that we're adding conditional rendering based on the engine name, not on its parameters — this will quickly become unmaintainable as we add more databases. 

Maybe we could create separate files for custom DBs like BQ, GSheets?

Reply:
Ha, I didn’t realize that we both made the same comment. :)